instrumentation/resources/library/src/test/java/io/opentelemetry/instrumentation/resources/ContainerResourceTest.java

```diff
@@ -50,4 +54,39 @@ void bothVersionsFail() {
     Resource resource = containerResource.buildResource();
     assertThat(resource).isSameAs(Resource.empty());
   }
+
+  @Test
+  void testAlternateEncoding() throws Exception {
+    String containerId = "ac679f8a8319c8cf7d38e1adf263bc08d231f2ff81abda3915f6e8ba4d64156a";
+    String line = "13:name=systemd:/podruntime/docker/kubepods/" + containerId + ".aaaa";
+    Charset ibmCharset = Charset.forName("Cp1047");
+    byte[] utf8 = line.getBytes(StandardCharsets.UTF_8);
+    byte[] ibm = line.getBytes(ibmCharset);
+    assertThat(ibm).isNotEqualTo(utf8);
+
+    String ibmAsString = new String(ibm, ibmCharset);
+    // Different bytes, different encoding, same semantic string value
+    assertThat(line).isEqualTo(ibmAsString);
+
+    // Make temp file that contains the IBM encoding
+    File tempFile = File.createTempFile("tmp", "mountinfo");
+    tempFile.deleteOnExit();
+    try (FileOutputStream out = new FileOutputStream(tempFile)) {
+      out.write(ibm);
+    }
+    ContainerResource.Filesystem fs =
+        new ContainerResource.Filesystem() {
```

Comment:
I think we usually add a constructor that accepts a supplier that can be replaced for testing, see https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/60c7993bcb56dd1abd828dabb2ba0c68aba1a64e/instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/HostIdResource.java#L56

Reply:
Can do